src/plugins/intel_gpu/src/kernel_selector/kernels/rope/rope_kernel_opt.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "rope_kernel_base.h"
+
+namespace kernel_selector {
+class RoPEKernelOpt : public RoPEKernelBase {
+public:
+    using Parent = RoPEKernelBase;
+    RoPEKernelOpt() : RoPEKernelBase("rope_opt") {
+        vec_size = 1;
+    }
+    virtual ~RoPEKernelOpt() {}
+
+    KernelsData GetKernelsData(const Params& params) const override;
+    KernelsPriority GetKernelsPriority(const Params& params) const override;
+    ParamsKey GetSupportedKey() const override;
+protected:
+    JitConstants GetJitConstants(const rope_params& params, DispatchData dispatchData) const override;
+    DispatchData SetDefault(const rope_params& params) const override;
+private:
+    mutable size_t vec_size;
```

Comment:
We have a single object instance of this class used for all the Rope layers in the model. To prevent any issues with Rope having different data types or rotary_ndims (and using different vec_size), it's better to introduce some function like `size_t get_vec_size(...)` and call it directly from `GetJitConstants()` and `SetDefault()`